paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/PaymentSelection.kt

```diff
@@ -392,7 +385,7 @@ internal val PaymentSelection.paymentMethodType: String
         PaymentSelection.Link -> "link"
         is PaymentSelection.New.Card -> paymentMethodCreateParams.typeCode
         is PaymentSelection.New.GenericPaymentMethod -> paymentMethodCreateParams.typeCode
-        is PaymentSelection.New.LinkInline -> linkPaymentDetails.paymentMethodCreateParams.typeCode
```

Comment:
Can we simplify this when to just be `is PaymentSelection.New -> -> paymentMethodCreateParams.typeCode` and get rid of all the other `PaymentSelection.New.*` cases?